publisher/internal/analytics/ethereum/types.go

```diff
@@ -1,11 +1,27 @@
 package ethereum
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"time"
 
+	"github.com/SyntropyNet/swapscope/publisher/pkg/analytics"
 	"github.com/SyntropyNet/swapscope/publisher/pkg/repository"
+	"github.com/SyntropyNet/swapscope/publisher/pkg/types"
 )
 
+type Operation interface {
+	// Common methods shared by Addition and Removal
+	ExtractFromEventLogs(EventLog, *Analytics) (Operation, error)
+	PrintPretty() string
```

Comment:
Conform to `Stringer` interface by renaming `PrintPretty` to `String`.

Also, `Print*` implies it actually prints something somewhere(logs, screen, stream, etc).